students/semko_krzysztof/lesson_13_objects_and_classes/exercise_one.py

```diff
@@ -0,0 +1,94 @@
+"""
+Write a definition for a class named Circle with attributes
+center and radius, where center is a Point object and radius
+is a number.
+
+Instantiate a Circle object that represents a circle
+with its center at (150, 100) and radius 75.
+
+Write a function named point_in_circle that takes a Circle
+and a Point and returns True if the Point lies in or
+on the boundary of the circle.
+
+Write a function named rect_in_circle that takes a Circle
+and a Rectangle and returns True if the Rectangle lies
+entirely in or on the boundary of the circle.
+
+Write a function named rect_circle_overlap that takes
+a Circle and a Rectangle and returns True if any of the
+corners of the Rectangle fall inside the circle.
+Or as a more challenging version, return True if any
+part of the Rectangle falls inside the circle.
+"""
+
+import math
+
+
+class Point:
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+
+class Circle:
+    def __init__(self, x, y, radius):
+        self.center = Point(x, y)
+        self.radius = radius
+
+
+class Rectangle:
+    def __init__(self, width, height, corner):
+        self.width = width
+        self.height = height
+        self.corner = corner
+
+
+def point_in_circle(circle, point):
+    distance = math.sqrt(
+        pow(point.x - circle.center.x, 2) +
+        pow(point.y - circle.center.y, 2))
+    return circle.radius > distance
```

Comment:
Zgodnie z poleceniem:

> Write a function named rect_in_circle that takes a Circle and a Rectangle and returns True if the Rectangle lies entirely in or on the boundary of the circle.

Ta funkcja zwróci false dla: `point_in_circle(Circle(0, 0, 2), Point(2, 0))`.

Jest to błąd implementacyjny, który na pewno by wyszedł na unit testach gdyby zbiór testowy obejmował wartości graniczne.

Nie poprawiaj